public/app/features/variables/state/sharedReducer.ts

```diff
@@ -73,8 +73,20 @@ const sharedReducerSlice = createSlice({
       }
     },
     duplicateVariable: (state: VariablesState, action: PayloadAction<VariablePayload<{ newId: string }>>) => {
+      function escapeRegExp(string: string): string {
+        return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+      }
+
       const original = cloneDeep<VariableModel>(state[action.payload.id]);
-      const name = `copy_of_${original.name}`;
+      const originalName = original.name.match(/^copy_of_(.*)_\d+$/)?.[1] || original.name;
```

Comment:
I thought, this approach would be better than having `copy_of_copy_of_copy_of`, that's why I went for this implementation.

Reply:
Yes, `copy_of_query0_2` is a copy of `copy_of_query0_1`, but its name suggests otherwise - that it is actually a _second_ copy of `query0`. If I make a copy of `query0` **now**, it will be `copy_of_query0_3`, making them all seem origined from the same variable. 

Having  `copy_of_copy_of_copy_of` is just the way it will be if you're making a copy of a variable that has a name "copy_of".

Reply:
If `copy_of_query0_1` is a copy of `query0`, then `copy_of_query0_2` is transitively a copy of `query0`, too. :)
I'd say, that they are originating from the same variable, some just take a couple of steps to get there.

But if this behaviour isn't preferable, I will get the `copy_of_copy_of_` one back.
Would the correct new name for a copy of `copy_of_query0_1` be `copy_of_copy_of_query0_1_1`?

Reply:
I see it like this:
1. If I duplicate the variable with the name `varA` the name assigned will be `copy_of_varA`
2. If I duplicate `varA` again, the new variable name will be `copy_of_varA_1` because `copy_of_varA` already exists
3. If I duplicate `copy_of_varA`, the name is `copy_of_copy_of_varA` since it is a copy of the copy.

With this logic, we only use the numeric suffix when there is a name conflict.

Reply:
And if I duplicate `copy_of_varA_1`?

Reply:
> Sure, but you can change everything about `copy_of_query0_1` before making a copy of it - its type, its values, everything. And then it doesn't matter what it originates from.

At this point, a name change from `copy_of_` would also make a lot of sense. Otherwise, it doesn't tell you much. :)

Please, have a look at the updated behaviour.